### ISSUE TITLE
VidCoder slower than Handbrake

### ISSUE THREAD

**Issue by LeKeiser** [RandomEngy/VidCoder#1137]
### Problem Description

Hello,
I have both the latest version of VidCoder and HandBrake.
I'm riping a MKV file created with MakeMKV.
I'm using the basic configuration on both software. H265 10Bit and everything is set Same As Source. 
HandBrake converts the file in 57 minutes.
VidCoder starts at around 56 minutes, then after a few minutes, the estimate time goes up to 1h58... And it goes down reaaaaally slowly. 
I've uninstalled and re-installed VC, same thing.
It's really basically the same configuration. H265 10bit and the audio and video is the same as the source. 
I don't understand the estimate time difference.

Any idea ?

Thanx for your comment :)

### What version of VidCoder are you running?

8.25

**Comment by RandomEngy** [comment]
Usually when this happens it's a small difference in the encoding settings. Can you provide logs for both the HandBrake and VidCoder encodes?

**Comment by LeKeiser** [comment]
Will do :)
But just to make sire, how should I proceed ? 
Same source, I launch a rip with both software, I let it go till the end, and then I get the logs ? 
Or should I proceed another way ?

**Comment by RandomEngy** [comment]
Yes, same source, same settings, note how long it takes and provide the logs.

**Comment by LeKeiser** [comment]
So I riped the movie with VidCoder. 1h33min. A bit longer actually, but...
I got the log file for this rip, but it's 1530 lines. I guess you only need a portion of it, right? Could you tell me what you want to check? 
I'm riping the same movie right now with HandBrake. ETA 51min. I will add the log file when it's done. I guess you also need a portion of it?

Thanx for your help :)

**Comment by RandomEngy** [comment]
You can post the whole thing to pastebin. Or else, these are the interesting bits:

> [10:00:23] json job:
> {
>   "Audio": {
>     "AudioList": [],
>     "CopyMask": [
>     ...

> [10:00:23] job configuration:
> [10:00:23]  * source
> [10:00:23]    + F:\VIDEO_TS
> [10:00:23]    + title 1, chapter(s) 23 to 23
> ...

**Comment by LeKeiser** [comment]
Alright.
Here is the VidCoder log : https://pastebin.com/0WGf0f0k
And here is the HandBrake Log : https://pastebin.com/pbXCPRpD

I must have missed something in the configuration of VidCoder because the file it rendered is twice the size of the file HandBrake compressed... The overall bitrate is lower with Handbrake but the rest is similar 

HandBrake : https://pastebin.com/7mfYaxym
VidCoder : https://pastebin.com/jS54QPSa

Hope everything is right with my logs :)

**Comment by RandomEngy** [comment]
There's a lot that's different there. The biggest one is that the HandBrake encode is using x265 (software) and VidCoder is using NVEnc H265 (hardware). They are also using different profiles (fast vs medium).

**Comment by LeKeiser** [comment]
Sorry sorry sorry... I uploaded the wrong log. When I launched the rip, it was set up to use my NVidia card. I stopped it and used the "correct" settings : https://pastebin.com/ck8ABucP

`"Video": {
    "Encoder": "x265_10bit",
    "Level": "auto",
    "TwoPass": false,
    "Turbo": false,
    "ColorMatrixCode": 0,
    "Options": "strong-intra-smoothing=0:rect=0:aq-mode=1:rd=4:psy-rd=0.75:psy-rdoq=4.0:rdoq-level=1:rskip=2",
    "Preset": "slow",
    "Profile": "main10",
    "Quality": 22,
    "QSV": {
      "Decode": false
    },
    "HardwareDecode": 0`

I think this time I have the right log?

**Comment by LeKeiser** [comment]
Very weird.
I tried yesterday the hardware NVEnc H265, so see how fast it would encode a MKV.
But I did use the software option today. And I uploaded the log of the encoding.
I just launched it again now. 1h27min till the rip is done. Right now the log file is empty. Created, but 0 bytes. I guess  VidCoder will write it down when the rip is done?

**Comment by LeKeiser** [comment]
Now it should be alright :)
VidCoder : https://pastebin.com/uW9NJkVF
HandBrake : https://pastebin.com/pbXCPRpD

**Comment by RandomEngy** [comment]
With VidCoder you've chosen the "slow" preset and with HandBrake you've chosen the "fast" preset.

**Comment by LeKeiser** [comment]
Really ? I don't remember changing a setting called Preset on either of these programs. I don't even know where those settings are. But I'll check. 
What's the best setting? Fast or Slow ? What does it change ?
Is it the only difference between the two configurations?

And again thanx a lot for your help :)
